Renamed SDL_ttf.h to SDL2_ttf.h

ellisvelo opened the pull request:
Hello,

I ran into a small problem when c2ffi is trying to execute because it is looking for SDL2_ttf.h referenced in sdl2-ttf.asd, but the file doesn't exist. This pull request renames SDL_ttf.h to SDL2_ttf.h so that c2ffi executes successfully.  I wasn't sure if sdl2-ttf.asd should change or the header filename because some of the spec files start with SDL2 and others start with SDL. Thanks for looking into this.

ellisvelo committed:
Renamed SDL_ttf.h to SDL2_ttf.h

ellisvelo committed:
Define functions for getting and setting the font outline.

TTF_SetFontOutline outline argument represents the pixel size of the outline and
not a packed bit array like TTF_SetFontStyle.

Commit:
Updated for darwin aarch64 support

Commit:
Allow multiple external includes when c2ffi is invoked

justjoheinz commented:
@ellisvelo  I just ran into the same problem, wanted to make a PR and noticed that the maintainer did not do anything on github since 2020. I think it is time to fork these repositories. You already did an awesome job with the cl-sdl2 library and M1 architecture. Would you be willing to? 

I also would like to get https://github.com/TatriX/cl-sdl2-tutorial running again and link the tutorial to the cl-sdl2 repo, so that more people are enabled to write  graphical apps in Common Lisp.

ellisvelo commented:
Thanks for the compliment @justjoheinz. 👍  I do have a fork of this repo (https://github.com/ellisvelo/cl-sdl2-ttf), but I think that I should make a PR for the Sharp Lispers and combine efforts: https://github.com/sharplispers/cl-sdl2-ttf

That's a good idea about the `cl-sdl2-tutorial` because I think the LazyFoo tutorials have changed too. Go for it!

ellisvelo committed:
Address problem with using nreverse on a macro argument

Sometimes there is a compilation failure with sbcl because nreverse is
used on a macro argument.
